core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java

```diff
@@ -66,7 +69,7 @@
     private Duration infoUpdateInterval = new Duration(3, TimeUnit.SECONDS);
 
     private int writerCount = 1;
-    private int taskConcurrency = 16;
```

Comment:
It feels like we should have an upper limit here, otherwise there's a risk of running into small page related problems.

With the default partitioning buffer size of 32MB (https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java#L60) and the default hash partition count of 100 (https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java#L47) pages produced by the PartitionedOutputOperator  are only ~320KB large. Further partitioning into 16 partitions reduces page size to ~20KB what seems to be pretty low already.

Reply:
> PartitionedOutputOperator

`task.concurrency` doesn't really affect `PartitionedOutputOperator` operator produced page sizes.
Only `PartitioningExchanger` (local exchange) is affected.
I will cap it to 32 for now since it's what I've tested and it's what (32 cores) we recommend

btw: I'm planning to play with buffer sizes next